### PR TITLE
Set the right type for the default value in TableView

### PR DESCRIPTION
**Pull request by boutinb**
Fixes jasp-stats/jasp-test-release#1229

### CODE REVIEW

**Comment by Kucharssim** [comment]
Also fixes jasp-stats/jasp-test-release#1230

File: Desktop/widgets/tableviewbase.cpp [hunk 1]
```diff
@@ -153,3 +153,25 @@ void TableViewBase::termsChangedHandler()
 	if (_boundControl)
 		_boundControl->resetBoundValue();
 }
+
+QVariant TableViewBase::defaultValue() const
+{
+	// Force the QVariant to have the right type
+	if (itemType() == JASPControl::ItemType::Integer)
```
Comment (JorisGoosen) [inline Desktop/widgets/tableviewbase.cpp]: This should be a switch

**Review by JorisGoosen — CHANGES_REQUESTED** [review]

File: Desktop/widgets/tableviewbase.cpp (hunk 1 above)
Comment (boutinb) [reply Desktop/widgets/tableviewbase.cpp]: OK done

Comment (JorisGoosen) [reply Desktop/widgets/tableviewbase.cpp]: You should rerequest a review ;)

File: Desktop/widgets/tableviewbase.cpp [hunk 2]
```diff
@@ -153,3 +153,31 @@ void TableViewBase::termsChangedHandler()
...
+		if (_defaultValue.type() == QVariant::Double)	return _defaultValue;
+		if (_defaultValue.canConvert(QVariant::Double))	return _defaultValue.toDouble();
+		break;
+	}
+	case JASPControl::ItemType::String:
+	{
+		if (_defaultValue.type() == QVariant::String)	return _defaultValue;
+		if (_defaultValue.canConvert(QVariant::String))	return _defaultValue.toString();
+		break;
+	}
```
Comment (JorisGoosen) [inline Desktop/widgets/tableviewbase.cpp:179]: You could've added a `default` here but I think GCC doesnt trust the switches too much, so this is not so bad.

**Review by JorisGoosen — APPROVED** [review]

**Pull request merged**